hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedColumnFamilyOptions.java

```diff
@@ -36,6 +36,10 @@ public ManagedColumnFamilyOptions(ColumnFamilyOptions columnFamilyOptions) {
   @Override
   public ManagedColumnFamilyOptions setTableFormatConfig(
       TableFormatConfig tableFormatConfig) {
+    // Close the previous TableFormatConfig before overwriting.
+    if (tableFormatConfig() instanceof ManagedBlockBasedTableConfig) {
+      ((ManagedBlockBasedTableConfig) tableFormatConfig()).close();
+    }
```

Comment:
It is better to throw an exception for overwriting.  We may throw an UnsupportedOperationException.
```
    final TableFormatConfig previous = tableFormatConfig();
    if (previous != null) {
      throw new UnsupportedOperationException("Overwrite is not supported");
    }
```
Or we may enforce close before overwrite.
```
    if (previous instanceof ManagedBlockBasedTableConfig) {
      if (!previous.isClose()) { // isClose() is a new method
        throw new IllegalArgumentException("Overwriting an unclosed value");
      }
    } else if (previous != null) {
      throw new UnsupportedOperationException("Overwrite is not supported for " + previous.getClass());
    }
```